### DISCUSSION TITLE
Questions About Nested Relational Data

### DISCUSSION THREAD

**Discussion by erondpowell** [directus/directus#17165] [Archive]
As I'm finishing my first re-edit of the API section on nested relational data, a few questions linger:

- Is it possible to delete a relationally-linked item _and delete nested relational item(s)_ at the same time? How do we do that?

- For M2Ms, is it possible to update multiple nested items with unique values?

**Comment by rijkvanzanten** [comment]
> Is it possible to delete a relationally-linked item and delete nested relational item(s) at the same time? How do we do that?

You can configure the relationship to auto delete the linked item on deletion from the relationship settings of the field. There's no syntax in the API to mark a related item for deletion.

> For M2Ms, is it possible to update multiple nested items with unique values?

Yeah! Here's an example of both syntaxes:

```json
{
  "m2m_field": [
    { "id": 15, "title": "Updated value 1" },
    { "id": 31, "title": "Updated value 2" }
  ]
}
```

```json
{
  "m2m_field": {
    "update": [
      { "id": 15, "title": "Updated value 1" },
      { "id": 31, "title": "Updated value 2" }
    ]
  }
}
```

The trick is to include the existing primary key (in this case `id`). If you omit the primary key, a new item will be created.

**Reply by erondpowell** [reply]
@rijkvanzanten  Okay thanks!  

> There's no syntax in the API to mark a related item for deletion.

What's the reasoning behind not being able to delete nested relational items from the the API? Safety reasons?

**Reply by rijkvanzanten** [reply]
No reason in particular 🤷🏻 Hasn't been requested/built before
